src/output-json-mqtt.c

```diff
@@ -60,7 +61,9 @@ typedef struct LogMQTTLogThread_ {
 
 bool JsonMQTTAddMetadata(void *vtx, JsonBuilder *js)
 {
-    return rs_mqtt_logger_log(vtx, MQTT_DEFAULTS, js);
+    // we pass 0 here as max loggable message length to ensure the full message
```

Comment:
I think this is what we want to avoid :-/

Reply:
Note that this function is only used for metadata added to alerts, not for regular metadata logging. This analogous to how it's done, for example, in the Postgres parser.
I *think* I could change this to also use the config for the `mqtt` type, but it might be a bit complicated to store this configuration in the corresponding context. This would need some more digging.

Reply:
If we are triggering an alert, isn't it better to be able to see what's in the packet, in full?

Reply:
True. Depends on the alert, I guess.
It looks like there is some common pattern here:

* For potentially large transactions (like HTTP, FTP, ...) the metadata logged with an alert only contains "small" metadata (headers, user agent, ...) while the contents of the body can only be gathered partially from the `packet` and `payload` fields. You never get the full body up to the alert logged anywhere in the alert (you would need pcap logging for that, or file extraction).
* For small payloads (like DNS) the metadata logged with the alert contains pretty much the whole parsed content of the packet/payload.

I wonder which one MQTT fits better into... I would always have expected it to fall into the second category, but now that I think about it it could also be the former.

Reply:
BTW I looked into how we can ensure that the `msg-log-limit` setting in the `mqtt` EVE output entry is also used for outputting the alert metadata. These arbitrary values for password logging and log limitation were picked because we can't get to the `mqtt` logger thread`s `ThreadVars` in which the setting for the currently configured output are stored.

I can imagine it might make sense here to extend `EveJsonSimpleTxLogFunc` and `EveJsonSimpleAppLayerLogger` so they can also carry config state, as well as make it possible to pass some config state as a pointer or function which has been parsed in one spot from `output-json-*.c` via `OutputRegisterTxSubModule`. Any ideas? It's not that straightforward since we can possibly have multiple EVE outputs with different settings.
(But that discussion is maybe a bit more involved than a random code comment thread ;))

Reply:
See #11160 for another approach. Until this is sorted I'll use 1KB here to get this going.